q2_diversity/_procrustes.py

```diff
@@ -114,3 +117,101 @@ def _procrustes_monte_carlo(reference: np.ndarray, other: np.ndarray,
                       index=pd.Index(['results'], name='id'))
 
     return df
+
+
+def partial_procrustes(reference: OrdinationResults, other: OrdinationResults,
+                       pairing: qiime2.CategoricalMetadataColumn,
+                       dimensions: int = 5) -> OrdinationResults:
+    if reference.samples.shape[1] < dimensions:
+        raise ValueError('Cannot fit fewer dimensions than available')
+
+    if other.samples.shape[1] < dimensions:
+        raise ValueError('Cannot fit fewer dimensions than available')
+
+    pairing = pairing.to_series()
+    pairing = pairing[~pairing.isnull()]
+
+    if len(pairing) == 0:
+        raise ValueError('The metadata are lacking paired samples')
+
+    ref_pairs = sorted(set(pairing.index) & set(reference.samples.index))
+    other_pairs = sorted(set(pairing.index) & set(other.samples.index))
```

Comment:
It is not assured that the set of samples represented here are the same, if for instance, one ordination lacked a sample. 

```suggestion
    pairing = pairing.to_series()
    pairing = pairing[~pairing.isnull()]
    pairing = pairing.loc[set(reference.samples.index) & set(other.samples.index)]

    if len(pairing) == 0:
        raise ValueError('The metadata are lacking paired samples')

    ref_pairs = sorted(set(pairing.index) & set(reference.samples.index))
    other_pairs = sorted(set(pairing.index) & set(other.samples.index))
```